### ISSUE TITLE
Header "types" are lost

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/openapi-extractor#109]
```json

**Comment by nickvergessen** [comment]
I just realised that typing them as `int` makes them int/number in the specs. But this is "not correct" as they are "strings", just with a numeric value?
Or are OpenAPI clients supposed to parse/cast the values internally?

**Comment by provokateurin** [comment]
Any type is supported in the headers. When you set the headers on the response you just need to make sure to not cast the value to string first. In most places in the server and apps this happens, but it can be changed and shouldn't be a breaking change since the serialization will lead to the same value.

> But this is "not correct" as they are "strings", just with a numeric value?

I don't understand what you mean. Any header value has to be serialized as a string, so everything is technically a string. If you cast the value to string in the app code then you are forced to type it as `string` or `numeric-string`, but you can also not cast it and let the AppFramework handle it.

**Comment by provokateurin** [comment]
One example already present in Talk is https://github.com/nextcloud/spreed/blob/0afabdecf29f700e038755c95ab59dc202a2c472/openapi.json#L11332 and in server https://github.com/nextcloud/server/blob/9afcec721fdc75c876bb62cc2de8295280ed4810/core/openapi.json#L5609.

**Comment by nickvergessen** [comment]
So if claiming a type while ^ makes it part of a combined string, then I will proceed doing so

**Comment by provokateurin** [comment]
Yeah just be careful to not cast when you don't want it. Then this issue can be closed because everything works as expected? Maybe the docs could be adjusted to state the support for non-string headers.
